### ISSUE TITLE
SASS do not support a @import statement 

### ISSUE THREAD

**Issue by mikeifomin** [christianalfoni/webpack-bin#144]
See example: http://www.webpackbin.com/N1Az33oG-

```

Module build failed: File to import not found or unreadable: /api/sandbox/14639973780838080/a.scss
Parent style sheet: stdin
      in /api/sandbox/14639973780838080/lib.scss (line 1, column 1)

```

Why?
That is because in [sass-loader](https://github.com/jtangelder/sass-loader/) do not working properly with a memory-fs. 
Sass works with the node-sass that [supports](https://github.com/sass/node-sass#importer--v200---experimental) the @import statement  via `importer` function, which return a filename **or** contents. In sass-loader the `importer` [return](https://github.com/jtangelder/sass-loader/blob/v3.2.0/index.js#L182) the filename and then node-sass(via LibSass) is trying to read from a physical file system. Of course 

So technically this is sass-loader bug but at the crossroads the memory-fs.

**Comment by christianalfoni** [comment]
Yeah, this is a limitation currently. Other libs also causes this constraint. Kinda sucks, but... hm... I suppose it would be possible to make **sass-loader** use the Webpack reference to the filesystem instead of hardcoded to _fs_? So whatever filesystem is put on webpack is what is being used by sass-loader?

**Comment by christianalfoni** [comment]
Thanks for the research btw :-)

**Comment by mikeifomin** [comment]
Im so interested in fixing this feature-bug that I think to write PR in a few days. But Im a new in an internal API of the Webpack so it can take match more time)
Maybe there is some simple string to fix it?

**Comment by christianalfoni** [comment]
My initial thought here is to fix the issue in sass-loader, so instead of using `require('fs')` it would point to the compiler instance filesystem. As you can see here it is changed: https://github.com/christianalfoni/webpack-bin/blob/master/server/sessionBundler.js#L82.

So if sass-loader is able to reference the compiler it should use that filesystem. By default the compiler uses `require('fs')`. 

Would that make sense?

**Comment by mikeifomin** [comment]
The last 4 hours of research I'v understand some of Webpack internals.
However problem is in the sass-loader only.

Node-sass has [an importer api](https://github.com/sass/node-sass#importer--v200---experimental) and a result callback take an object with:

```
file (String) - an alternate path for LibSass to use OR
contents (String) - the imported contents (for example, read from memory or the file system)
```

The sass-loader use the first one option [(direct link to source)](https://github.com/jtangelder/sass-loader/blob/v3.2.0/index.js#L182):

``` javascript
     done({
           file: resolvedFilename.replace(matchCss, '')
     });
```

So the LibSass takes a value from `file` ( an **absolute** url of a file need to be imported) and load via C api (LibSass is C library).

The easiest way to fix it is to use `contents` with data loaded by loader module 

``` javascript
     function readFileFromWebpackFileSystem(_path){
          // ???
     }
     done({
           contents: readFileFromWebpackFileSystem( resolvedFilename.replace(matchCss, '') )
     });
```

**Comment by mikeifomin** [comment]
How to get an access to a Webpack instance from a loader function?

**Comment by christianalfoni** [comment]
Ah, okay! Hm, I have not written any loaders for Webpack, but if you log out `this` inside the loader callback there might be some reference to the webpack instance there. Or I think an issue is needed to get some help on it

**Comment by mikeifomin** [comment]
Yeah! With `this._compiler.inputFileSystem` the sass-loader working perfect.)

**Comment by christianalfoni** [comment]
Ah, awesome, maybe you could give them a pull on that. It should really work like that :-)

**Comment by mikeifomin** [comment]
Of course! But the PR is not ready due to some refactoring of a main app I developing. My estimation of the pull is about 8 hours.
I think more convenient is to pull directly to the sass-loader. So approve may take some time.

**Comment by christianalfoni** [comment]
Yeah, I think pull directly to sass-loader is the best. It might be pushed out as a patch version though, as it does not really change anything... but, hm, probably take some time yes ;-)

But really great that we figured this one out! 👍

**Comment by mikeifomin** [comment]
Now finally I've created a [fork](https://github.com/mikeifomin/sass-loader/tree/memory-fs) and commit changes there. 
To be honest it will be better to add some tests with the memory-fs. The only test I've made is to put this string in my project and run it :grimacing:

```
  "sass-loader": "git@github.com:mikeifomin/sass-loader.git#memory-fs",
```

**Comment by christianalfoni** [comment]
Heh, yeah, that would be really great... though quite a bit of work. But always good to get feedback on the suggestion first anyways, author will just require tests if needed :)
